kafka/src/main/java/io/micronaut/configuration/kafka/processor/ConsumerState.java

```diff
@@ -427,7 +427,8 @@ protected void delayRetry(int currentRetryCount, Set<TopicPartition> partitions)
 
     protected boolean shouldRetryException(Throwable e, ConsumerRecords<?, ?> consumerRecords, ConsumerRecord<?, ?> consumerRecord) {
         if (info.errorStrategy.isConditionalRetry()) {
-            return kafkaConsumerProcessor.shouldRetryMessage(consumerBean, wrapExceptionInKafkaListenerException(e.getMessage(), e, consumerRecords, consumerRecord));
+            return kafkaConsumerProcessor.shouldRetryMessage(consumerBean, wrapExceptionInKafkaListenerException(e.getMessage(), e, consumerRecords, consumerRecord)) ||
+                info.exceptionTypes.stream().anyMatch(e.getClass()::equals);
```

Comment:
Ahh, what I had in mind was...

If `exceptionTypes` is set with one of the conditional retry strategies, then only invoke the conditional behaviour handler if the exception is a match. But now I realize that just doesn't even make sense, because then there's no way to specify that exceptions that *don't* match should just be an automatic retry.

Your version makes more sense. Great!